src/gui/sdl_mapper.cpp

```diff
@@ -2430,15 +2414,22 @@ static void CreateDefaultBinds() {
 	LOG_MSG("MAPPER: Loaded default key bindings");
 }
 
-void MAPPER_AddHandler(MAPPER_Handler * handler,MapKeys key, Bitu mods,char const * const eventname,char const * const buttonname) {
+void MAPPER_AddHandler(MAPPER_Handler *handler,
+                       SDL_Scancode key,
+                       uint32_t mods,
+                       const char *event_name,
+                       const char *button_name)
+{
 	//Check if it already exists=> if so return.
-	for(CHandlerEventVector_it it=handlergroup.begin(); it != handlergroup.end(); ++it)
-		if(strcmp((*it)->buttonname,buttonname) == 0) return;
+	for (const auto &handler_event : handlergroup) {
+		if (handler_event->button_name == button_name)
+			return;
+	}
 
 	char tempname[17];
 	safe_strcpy(tempname, "hand_");
-	safe_strcat(tempname, eventname);
-	new CHandlerEvent(tempname,handler,key,mods,buttonname);
+	safe_strcat(tempname, event_name);
+	new CHandlerEvent(tempname, handler, key, mods, button_name);
```

Comment:
Not sure if it's possible, but would be nice to make **MAPPER_AddHandler**'s `button_name`  argument also a  `std::string` similar to what was done in `CHandlerEvent`.